### PR TITLE
Feature di fragments

### PR DESCRIPTION
**Pull request by janeiromeosa**
Wondering if I should store Lon and Lat coordinates to a dataclass

### CODE REVIEW

File: app/src/main/java/com/prayertime/BaseActivity.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import androidx.core.app.ActivityCompat
+import com.google.android.gms.location.FusedLocationProviderClient
+import dagger.android.support.DaggerAppCompatActivity
+
+
+abstract class BaseActivity : DaggerAppCompatActivity() {
+
+    lateinit var fusedLocationClient: FusedLocationProviderClient
+
+    val PERMISSION_ID = 42
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/BaseActivity.kt]: add it inside a companion

File: app/src/main/java/com/prayertime/utils/LocationUtils.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        fusedLocationProviderClient!!.lastLocation
+            .addOnSuccessListener {loc: Location? ->
+                location.value = loc
+
+            }
+
+        return location
+    }
+
+}
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/utils/LocationUtils.kt]: @janeiromeosa does this class is used anywhere?

File: app/src/main/java/com/prayertime/view/location/LocationViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        compositeDisposable.add(
+            repository.getLocationInformation(lat, lon)
+                .doOnSubscribe { progressObservable.postValue(true) }
+                .doOnError { progressObservable.value = false }
+                .subscribe({ value ->
+                    locationObservable.value = value
+                    progressObservable.value = false
+                }, { errorObservable.value = true })
+        )
+    }
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/view/location/LocationViewModel.kt]: @janeiromeosa I would suggest to change the method name to `setlastKnowLocation(lat: String, lon: String)`. In this method just set the location of the user. Dont worry about data caching for now. We will make use of repository in some other method.

File: app/src/main/java/com/prayertime/view/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+class MainActivity : BaseActivity() {
+
+
+    private val locationViewModel by viewModel<LocationViewModel>()
+
+    @Inject
+    lateinit var providerFactory: ViewModelFactory
+
+
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/view/MainActivity.kt]: Please remove extra lines from the code.

File: app/src/main/java/com/prayertime/utils/LocationUtils.kt (hunk 2 above)
Comment (janeiromeosa) [reply app/src/main/java/com/prayertime/utils/LocationUtils.kt]: Nope. I was just testing out something. I will take it out.

File: app/src/main/java/com/prayertime/view/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        setContentView(R.layout.activity_main)
+
+        val navHostFragment = supportFragmentManager?.findFragmentById(R.id.nav_host_fragment) as NavHostFragment?
+        NavigationUI.setupWithNavController(bottom_navigation, navHostFragment!!.navController)
+
+        Toast.makeText(this@MainActivity, "MainActivity", Toast.LENGTH_LONG).show()
+
+
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+//        locationViewModel.getLocation(fusedLocationClient)
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/view/MainActivity.kt]: I think these two are not required here,

File: app/src/main/java/com/prayertime/view/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    }
+
+    @SuppressLint("MissingPermission")
+    fun getFusedLocation(){
+        fusedLocationClient!!.lastLocation
+            .addOnSuccessListener {loc: Location? ->
+//                location.value = loc
+            }
+    }
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/view/MainActivity.kt]: This one is also not required

File: app/src/main/java/com/prayertime/view/direction/DirectionActivity.kt [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package com.prayertime.view.direction
+
+class DirectionActivity {
+
+}
```
Comment (geek2driod) [inline app/src/main/java/com/prayertime/view/direction/DirectionActivity.kt:5]: I guess we will not use this activity also, as we are using fragments.

**Pull request merged**